wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -1494,6 +1495,10 @@ class KotlinGenerator private constructor(
 
   private val Field.typeName: TypeName
     get() {
+      if (type == ProtoType.STRUCT_MAP) return type!!.typeName
+      if (type == ProtoType.STRUCT_LIST) return type!!.typeName
+      if (type == ProtoType.STRUCT_VALUE) return type!!.typeName.copy(nullable = true)
+      if (type == ProtoType.STRUCT_NULL) return type!!.typeName.copy(nullable = true)
```

Comment:
Maybe we could change their encodeMode?
Something like `STRUCT_MAP` -> `MAP`, `STRUCT_LIST` -> `REPEATED` (not packed? can it be packed?), then `STRUCT_VALUE` and `STRUCT_NULL` would be `OMIT_IDENTITY`

Reply:
Does that also eliminate the special casing for identity value and accepts null below? Because if so, i'm supportive.

Reply:
will follow up with that